### PR TITLE
Add explicit softmax dimension

### PR DESCRIPTION
**Pull request by notnami**
This explicitly specifies the softmax dimension to maintain compatibility with newer versions of PyTorch

### CODE REVIEW

**Comment by notnami** [comment]
Looks like Travis is failing because older versions of PyTorch did not support passing an explicit argument for dimension. I don't know if there are reasons for this project to maintain support for those versions, but if there isn't, it might make sense to upgrade.

**Comment by maksymbevza** [comment]
Could you please also fix it everywhere, e.g. for the fully-convolutional model as well?

**Comment by JianyuZhan** [comment]
Yep, the latest `pytorch 0.3.0` breaks some compatibility, and this problem is just among one of them - a warning.  I think we prefer to wait for things settling down before updating to accommodate `0.3.0`. Currently Travis environment is pinning the pytorch to `0.2.0`.

**Comment by srush** [comment]
Hmm, I am fine with switching to 0.3, but maybe we can have a workaround for 0.2 in the meantime.

**Comment by pltrdy** [comment]
The best practice would probably be to open a branch for 0.3 development and keep 0.2 on master, what do you think?

**Comment by srush** [comment]
if it is just a warning, I don't think we need to do anything. What other issues are there? Can we turn on travis for 0.3?

**Comment by JianyuZhan** [comment]
@srush, I have enabled the pytorch 0.3.0 during the IO seperation work, it reports at least two errors, besides this warning.  And one of the error I am afraid it is not our fault. But I agree we could make a temporary branch for it. I will investigate this after the preprocess work.

**Comment by srush** [comment]
finally got this checked in
